src/test/java/io/percy/selenium/SdkTest.java

```diff
@@ -108,4 +112,46 @@ public void snapshotWithOptions() {
     options.put("widths", Arrays.asList(768, 992, 1200));
     percy.snapshot("Site with options", options);
   }
+
+    @Test
+    public void takeScreenshotWhenNonRemoteWebDriver() {
+      assertThrows(UnsupportedOperationException.class, () -> {
+        percy.screenshot("Test");
+      });
+    }
+  @Test
+  public void takeScreenshot() {
+    RemoteWebDriver mockedDriver = mock(RemoteWebDriver.class);
+    HttpCommandExecutor commandExecutor = mock(HttpCommandExecutor.class);
+    try {
+      when(commandExecutor.getAddressOfRemoteServer()).thenReturn(new URL("https://hub-cloud.browserstack.com/wd/hub"));
+    } catch (Exception e) {
+    }
+    percy = new Percy(mockedDriver);
+    when(mockedDriver.getSessionId()).thenReturn(new SessionId("123"));
+    when(mockedDriver.getCommandExecutor()).thenReturn(commandExecutor);
+    DesiredCapabilities capabilities = new DesiredCapabilities();
+    capabilities.setCapability("browserName", "Chrome");
+    when(mockedDriver.getCapabilities()).thenReturn(capabilities);
+    percy.screenshot("Test");
+  }
+
+    @Test
+    public void takeScreenshotWithOptions() {
+      RemoteWebDriver mockedDriver = mock(RemoteWebDriver.class);
+      HttpCommandExecutor commandExecutor = mock(HttpCommandExecutor.class);
+      try {
+        when(commandExecutor.getAddressOfRemoteServer()).thenReturn(new URL("https://hub-cloud.browserstack.com/wd/hub"));
+      } catch (Exception e) {
+      }
+      percy = new Percy(mockedDriver);
+      when(mockedDriver.getSessionId()).thenReturn(new SessionId("123"));
+      when(mockedDriver.getCommandExecutor()).thenReturn(commandExecutor);
+      DesiredCapabilities capabilities = new DesiredCapabilities();
+      capabilities.setCapability("browserName", "Chrome");
+      when(mockedDriver.getCapabilities()).thenReturn(capabilities);
+      Map<String, Object> options = new HashMap<String, Object>();
+      options.put("scope", "div");
+      percy.screenshot("Test", options);
```

Comment:
Shouldn't there be some `expectation` here?

Reply:
This is still not sure. what are the options poa is going to support will come as a part of second. So just blankly passing it.

Reply:
I meant we need some expectation after `percy.screenshot` to verify it worked correctly. Maybe check if it hit the correct endpoint or something else

Reply:
Well this is tricky since we can only expect either result of function or calls can be expected when they are mocked. now mocking api call will need lot of restructuring because we need to pass mocked httpclient into function call?

can you suggest any other thing we can check?

Reply:
Resolved